salt/states/boto_cloudfront.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+'''
+Manage CloudFront distributions
+
+.. versionadded:: develop
+
+Create, update and destroy CloudFront distributions.
+
+This module accepts explicit AWS credentials but can also utilize
+IAM roles assigned to the instance through Instance Profiles.
+Dynamic credentials are then automatically obtained from AWS API
+and no further configuration is necessary.
+More information available `here
+<https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html>`_.
+
+If IAM roles are not used you need to specify them,
+either in a pillar file or in the minion's config file:
+
+.. code-block:: yaml
+
+    cloudfront.keyid: GKTADJGHEIQSXMKKRBJ08H
+    cloudfront.key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+
+It's also possible to specify ``key``, ``keyid``, and ``region`` via a profile,
+either passed in as a dict, or a string to pull from pillars or minion config:
+
+.. code-block:: yaml
+
+    myprofile:
+        keyid: GKTADJGHEIQSXMKKRBJ08H
+        key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+        region: us-east-1
+
+.. code-block:: yaml
+
+    aws:
+        region:
+            us-east-1:
+                profile:
+                    keyid: GKTADJGHEIQSXMKKRBJ08H
+                    key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+                    region: us-east-1
+
+:depends: boto3
+'''
+
+# Import Python Libs
+from __future__ import absolute_import
+import difflib
+import logging
+
+import yaml
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load if boto is available.
+    '''
+    if 'boto_cloudfront.get_distribution' in __salt__:
+        return 'boto_cloudfront'
+    return False
```

Comment:
Same comment here: we should return `False` with a comment explaining why the module didn't load to help out the user.